Ubuntu 22.04

R3CK0 opened the pull request:
Modified all tests to no longer require on indy-sdk. Started from where Adam left off. Please correct any changes where I am using wrong inputs/outputs

burdettadam committed:
feat: upgrade sdk to vdr

Signed-off-by: Adam Burdett <burdettadam@gmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

burdettadam committed:
feat: upgrade sdk_wallet_handle fixture

Signed-off-by: Adam Burdett <burdettadam@gmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

burdettadam committed:
feat: update sdk_wallet_handle calls

Signed-off-by: Adam Burdett <burdettadam@gmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

burdettadam committed:
feat: upgrade sdk to vdr in setup

Signed-off-by: Adam Burdett <burdettadam@gmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

R3CK0 committed:
finnished refactoring all tests containing libindy replaced with new
packages starting debugging process with tests

Signed-off-by: Nicholas Massad <nicholasmassad@hotmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

R3CK0 committed:
modified event-loop to handle module scope in tests

Signed-off-by: Nicholas Massad <nicholasmassad@hotmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

WadeBarnes commented on setup.py:
```diff
@@ -136,6 +135,9 @@ def run(self):
                         'sortedcontainers==2.1.0',
                         ### Tests fail without version pin (GHA run: https://github.com/udosson/indy-plenum/actions/runs/1078741118)
                         'ujson==1.33',
+                        'indy_vdr==0.4.2', 
+                        'aries-askar==0.4.3', 
+                        'indy-credx==1.1.1'
```
This makes the indy-plenum deployment package dependent on client side components that are only used for tests.

WadeBarnes reviewed (changes requested):
Since this is based

R3CK0 committed:
removed libindy from dockerfile, modified request type to handle in test

Signed-off-by: Nicholas Massad <nicholasmassad@hotmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

R3CK0 committed:
modified devcontainer to autodetect installed python libraries

Signed-off-by: Nicholas Massad <nicholasmassad@hotmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

WadeBarnes committed:
Ensure deb packages are built for indy-vdr, aries-askar, and, indy-credx

Signed-off-by: Wade Barnes <wade@neoterictech.ca>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

WadeBarnes reviewed (commented):
General comment on the latest commit ...

It feels like there's duplication in the request parsing/conversion that could be consolidated.

WadeBarnes reviewed (changes requested):
@rhofirAtos, Your latest commit is missing DCO sign-off.

WadeBarnes commented:
> @rhofirAtos, Your latest commit is missing DCO sign-off.

lol, I see you noticed.  Thanks

safio commented:
@WadeBarnes I've addressed your comment regarding duplication in the request parsing/conversion by creating a helper function called convert_indy_vdr_request_to_plenum(). I'll include this in my next push.

WadeBarnes commented:
Tests are still having issues with `ioflo`.  It should be replaced as a `test` dependency so the tests can proceed and we get a better idea of where we're at with this PR.

WadeBarnes commented on setup.py:
```diff
@@ -99,7 +99,7 @@ def run(self):
                         # pinned because issue with fpm from v4.0.0
                         'importlib_metadata==3.10.1',
-                        # 'ioflo==2.0.2',
+                        'ioflo==2.0.3',
```
This will work, in that the expected version of `ioflo` will be installed, however `ioflo` is a dev/test dependency and not a runtime dependency.  Therefore it should not be defined here.  It should be defined in the `tests_require` section starting on line 30.

WadeBarnes commented:
DCO sign-off was not included in the most recent commit.  @rhofirAtos, please amend your latest commit with DCO sign-off.

WadeBarnes commented on .devcontainer/Dockerfile:
```diff
@@ -73,20 +73,19 @@ RUN apt-get update -y && apt-get install -y \
 
 RUN pip3 install -U \
+    # Required by newer verion of python
+    ioflo==2.0.3 \
```
This should be unnecessary.  `ioflo` should be installed as a dev dependency when plenum is installed in teh dev container by `"postCreateCommand": "pip install .[tests]",` in the devcontainer.json file.

R3CK0 committed:
Undid previous modification and modified testing functionnalities to
take into account and integrate new types, inputs and outputs from
indy-vdr

Signed-off-by: Nicholas Massad <nicholasmassad@hotmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

R3CK0 committed:
returned audit_multiple_ledger_in_one_batch to it's original state

Signed-off-by: Nicholas Massad <nicholasmassad@hotmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

R3CK0 committed:
added ioflo in stup.py

Signed-off-by: Nicholas Massad <nicholasmassad@hotmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

R3CK0 committed:
fixed audit ledger tests except multiple uncommitted node waiting for
response

- Fixed futures loop error
- Added Plenum request vs vdr request format
- Changed Submit_request to submit_action

Signed-off-by: Nicholas Massad <nicholasmassad@hotmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

R3CK0 committed:
Fixed tests in audit_ledger batching_3pc, bls, checkpoints

- Ongoing issue with client test: The build request in indy-vdr does not
  allow invalid protocol version to be set with is preventing the
  ability to verify invalid request do to protocol version in plenum

- Increased the timeout in some functions do to machine limitations
  causing tests to timeout and fail prematurely this needs to be tweaked
  as some tests require the timeout to be shorter and some longer. Need
  to find that sweet spot

Signed-off-by: Nicholas Massad <nicholasmassad@hotmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

R3CK0 committed:
removed inexistant function causing crash

Signed-off-by: Nicholas Massad <nicholasmassad@hotmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

R3CK0 committed:
fixed tests in signing
reintroduced indy-sdk to test invalid signatures that are prevented in
vdr
modified symbols from vdr functions to from sdk to vdr

Signed-off-by: Nicholas Massad <nicholasmassad@hotmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

R3CK0 committed:
readded sdk functionnalities

Signed-off-by: Nicholas Massad <nicholasmassad@hotmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

R3CK0 committed:
changed fixture signature from sdk to vdr and functions that utilise the
vdr format.

Signed-off-by: Nicholas Massad <nicholasmassad@hotmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

R3CK0 committed:
added sdk functionnalities to conftest fixtures and added helper
functions in pool_transaction and node_request to faciliate them

Signed-off-by: Nicholas Massad <nicholasmassad@hotmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

R3CK0 committed:
fixed signing tests

Signed-off-by: Nicholas Massad <nicholasmassad@hotmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

R3CK0 committed:
fixed validator_info tests

Signed-off-by: Nicholas Massad <nicholasmassad@hotmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

R3CK0 committed:
fixed monitoring tests

Signed-off-by: Nicholas Massad <nicholasmassad@hotmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

R3CK0 committed:
fixed forced requests test. Needed to use sdk since vdr request body
cannot be changed

Signed-off-by: Nicholas Massad <nicholasmassad@hotmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

R3CK0 committed:
fixed last broken test in signing

- When switching back to SDK reintroduced errors that where fixed in VDR

Signed-off-by: Nicholas Massad <nicholasmassad@hotmail.com>
Signed-off-by: NicholasMassad <nicholasmassad@hotmail.com>

WadeBarnes reviewed (changes requested):
@R3CK0 Please exclude the error logs from the commits.

Also the build is failing due to linting errors.

WadeBarnes commented:
@R3CK0, Whatever's happened between this point https://github.com/hyperledger/indy-plenum/actions/runs/14197083841 and now is a step backwards.  The tests where running and one slice was passing.  In the current state the tests aren't able to even run.

WadeBarnes commented:
That said, affected tests are running and passing locally.

WadeBarnes commented:
I'm able to recrate the error locally by running the tests like the workflows do:
```
RUSTPYTHONASYNCIODEBUG=0 python3 runner.py --pytest "python3 -m pytest -l -vv" --dir "plenum" --output "test-result-plenum-1.txt" --test-only-slice "1/8"
```

Logs get output to `test_list.txt`

```
==================================== ERRORS ====================================
___ ERROR collecting plenum/test/plugin/demo_plugin/test_plugin_removing.py ____
ImportError while importing test module '/workspaces/indy-plenum/plenum/test/plugin/demo_plugin/test_plugin_removing.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
plenum/test/plugin/demo_plugin/test_plugin_removing.py:5: in <module>
    from plenum.test.freeze_ledgers.helper import sdk_send_freeze_ledgers
E   ImportError: cannot import name 'sdk_send_freeze_ledgers' from 'plenum.test.freeze_ledgers.helper' (/workspaces/indy-plenum/plenum/test/freeze_ledgers/helper.py)
```

WadeBarnes commented:
Plenum slice tests are running again.

WadeBarnes committed:
Update repository references for Ursa.

- The Ursa package has been migrated to the Hyperledger repository.  The Sovrin repositories have been deprecated.
  - New location: https://hyperledger.jfrog.io/ui/native/indy/pool/bionic/master/u/ursa/
  - Previous location: https://repo.sovrin.org/deb/pool/bionic/master/u/ursa/

Signed-off-by: Wade Barnes <wade@neoterictech.ca>

WadeBarnes committed:
Update repository references for the indy-sdk.

- The indy-sdk packages have been migrated to the Hyperledger repository.  The Sovrin repositories have been deprecated.
  - New location: https://hyperledger.jfrog.io/ui/native/indy/pool/bionic/
  - Previous location: https://repo.sovrin.org/sdk/deb/pool/bionic/
    - Only the `master` and `stable` channels were migrated over.

Signed-off-by: Wade Barnes <wade@neoterictech.ca>

WadeBarnes commented:
I'll have a look at the conflicts shortly.

R3CK0 commented:
Removed the commit containing the error logs and the error identification script.

WadeBarnes commented:
> Removed the commit containing the error logs and the error identification script.

In doing so it appears you've also removed several commits I had made to the PR.

![image](https://github.com/user-attachments/assets/7547cd6d-e355-4413-8ba4-859ca7dac5a9)

WadeBarnes commented:
I'll put them back

R3CK0 commented:
I'm sorry, didn't realise, nothing else came up in my git logs after pulling, I assumed it was safe to push back one commit, or at least what I thought was one commit

WadeBarnes commented:
All fixed.

WadeBarnes committed:
Reorder the repository references to fix build issues.

Signed-off-by: Wade Barnes <wade@neoterictech.ca>
